crypto/sm2/sm2_sign.c

```diff
@@ -382,6 +382,7 @@ static int sm2_sig_verify(const EC_KEY *key, const ECDSA_SIG *sig,
         ret = 1;
 
  done:
+    BN_CTX_end(ctx);
     EC_POINT_free(pt);
     BN_CTX_free(ctx);
```

Comment:
The BN_CTX_end() call isn't required because it is freed here.

Reply:
Hmm, `BN_CTX_new.pod` says

```
in most cases `BN_CTX_end()` must be called before the `B<BN_CTX>` may be freed by `BN_CTX_free()`
```